play-json/shared/src/main/scala/play/api/libs/json/Reads.scala

```diff
@@ -98,14 +98,26 @@ object Reads extends ConstraintReads with PathReads with DefaultReads with Gener
 
   val path: PathReads = this
 
-  /** Returns a `JsSuccess(a)` (with root path) for any JSON value read. */
-  def pure[A](a: A): Reads[A] = Reads[A] { _ => JsSuccess(a) }
+  /**
+   * Returns a `JsSuccess(a)` (with root path) for any JSON value read.
+   *
+   * {{{
+   * import play.api.libs.json.Reads
+   *
+   * val r: Reads[String] = Reads.pure(f = "foo")
+   * // NOTE the required `f =` to pass as named parameter
+   * }}}
+   */
+  override def pure[A](f: => A): Reads[A] = Reads[A] { _ => JsSuccess(f) }
+
+  @deprecated("Use `pure` with `f:=>A` parameter", "2.7.0")
+  def pure[A](value: A): Reads[A] = Reads[A] { _ => JsSuccess(value) }
```

Comment:
So keeping the old version of `pure` around will help with binary compatibility, but it requires people to use `pure(f = ...)` to get the new version of `pure`?

Personally, I think it should be OK to remove the old version of `pure` and just use the new one, even if it's not binary compatible, since this change will happen for the next 'major' release of play-json anyway. For major releases, I think that source-compatibility is enough.

Reply:
I think we can do a 2.7.x release of play-json at some point. Not sure when though. WDYT @richdougherty?

What about using `private[json]` on this one to maintain bincompat?

Reply:
done